cpp/log/log_lookup-inl.h

```diff
@@ -68,8 +69,7 @@ void LogLookup<Logged>::UpdateFromSTH(const ct::SignedTreeHead& sth) {
     // handled more gracefully. E.g. we could retry a failed update
     // a number of times -- but until we know under which conditions
     // the database might fail (database busy?), just die.
-    CHECK_EQ(Database<Logged>::LOOKUP_OK,
-             db_->LookupByIndex(sequence_number, &logged))
+    CHECK(it->GetNextEntry(&logged))
```

Comment:
If we're committed to updating to that passed in STH and we don't have the data for it then we're screwed (since calls to this method happen via the new-sth-callback from the DB when we've decided to write out a new STH locally - so the data must be *somewhere*)